### PR TITLE
DBZ-1112 Strings.join() doesn't apply conversation for first element

### PR DESCRIPTION
**Pull request by jchipmunk**
Replace:
- Integer.parseInt() to Integer.valueOf()
- Short.parseShort() and new Short() to Short.valueOf()
- Long.parseLong() to Long.valueOf()

https://issues.jboss.org/browse/DBZ-1112

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java [hunk 1]
```diff
@@ -554,6 +554,26 @@ public JdbcConnection queryWithBlockingConsumer(String query, StatementFactory s
...
+     * Execute a SQL prepared query.
+     *
+     * @param preparedQueryString the prepared query string
+     * @param preparer the function that supplied arguments to the prepared statement; may not be null
+     * @param resultConsumer the consumer of the query results
+     * @return this object for chaining methods together
+     * @throws SQLException if there is an error connecting to the database or executing the statements
+     * @see #execute(Operations)
+     */
+    public JdbcConnection prepareQueryWithBlockingConsumer(String preparedQueryString, StatementPreparer preparer, BlockingResultSetConsumer resultConsumer)
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java]: Where is this one used?

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDefaultValuePreConverter.java [hunk 2]
```diff
@@ -161,7 +161,7 @@ private Object convertToDuration(Column column, String value) {
      * @return the converted value;
      */
     private Object convertToInteger(String value) {
-        return Integer.parseInt(value);
+        return Integer.valueOf(value);
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDefaultValuePreConverter.java]: What's the reasoning behind this change?

File: debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java (hunk 1 above)
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java]: Where is that one used?

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDefaultValuePreConverter.java (hunk 2 above)
Comment (jchipmunk) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDefaultValuePreConverter.java]: As you know `Integer.valueOf()`/`Short.valueOf()`/`Long.valueOf()` methods use cache and may return the cached value. I know Java compiler uses `*.valueOf()` methods for autoboxing, but if we already call `Integer.parseInt()`/`Short.parseShort()`/`Long.parseLong()` methods, then it is better to use `*.valueOf()` methods directly. Explicit is better than implicit.

Comment (gunnarmorling) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDefaultValuePreConverter.java]: Ok.

File: debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java (hunk 1 above)
Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java]: Ha, ok :)

**Pull request merged**